crates/sui-graphql-client/src/lib.rs

```diff
@@ -478,6 +502,159 @@ impl Client {
+            },
+        });
+
+        let response = self.run_query(&operation).await?;
+
+        if let Some(errors) = response.errors {
+            return Err(Error::msg(format!("{:?}", errors)));
+        }
+
+        let result: Option<DynamicFieldOutput> = response
+            .data
+            .and_then(|d| d.object)
+            .and_then(|o| o.dynamic_field)
+            .map(|df| df.into());
+        Ok(result)
+    }
+
+    /// Access a dynamic object field on an object using its name. Names are arbitrary Move values whose
+    /// type have copy, drop, and store, and are specified using their type, and their BCS
+    /// contents, Base64 encoded.
+    ///
+    /// This is the same as [`dynamic_object_field`] but instead of passing the bcs bytes, it takes
+    /// a value of a type that implements Serialize.
+    ///
+    /// This returns [`DynamicFieldOutput`] which contains the name, the value as json, and object.
+    pub async fn dynamic_object_field_with_name(
+        &self,
+        address: Address,
+        type_: &str,
+        name: impl Serialize,
+    ) -> Result<Option<DynamicFieldOutput>, Error> {
+        let bcs = bcs::to_bytes(&name).unwrap();
+        self.dynamic_object_field(address, type_, &bcs).await
+    }
+
+    /// Access a dynamic object field on an object using its name. Names are arbitrary Move values whose
+    /// type have copy, drop, and store, and are specified using their type, and their BCS
+    /// contents, Base64 encoded.
+    ///
+    /// This returns [`DynamicFieldOutput`] which contains the name, the value as json, and object.
+    pub async fn dynamic_object_field(
+        &self,
+        address: Address,
+        type_: &str,
+        bcs: &[u8],
+    ) -> Result<Option<DynamicFieldOutput>, Error> {
+        let operation = DynamicObjectFieldQuery::build(DynamicFieldArgs {
+            address,
+            name: crate::query_types::DynamicFieldName {
+                type_: type_.to_string(),
+                bcs: crate::query_types::Base64(base64ct::Base64::encode_string(bcs)),
+            },
+        });
+
+        let response = self.run_query(&operation).await?;
+
+        if let Some(errors) = response.errors {
+            return Err(Error::msg(format!("{:?}", errors)));
+        }
+
+        let result: Option<DynamicFieldOutput> = response
+            .data
+            .and_then(|d| d.object)
+            .and_then(|o| o.dynamic_object_field)
+            .map(|df| df.into());
+        Ok(result)
+    }
+
+    /// Get a page of dynamic fields for the provided address. Note that this will also fetch
+    /// dynamic fields on wrapped objects.
+    ///
+    /// This returns [`Page`] of [`DynamicFieldOutput`]s.
+    pub async fn dynamic_fields(
+        &self,
+        address: Address,
+        after: Option<String>,
+        before: Option<String>,
+        first: Option<i32>,
+        last: Option<i32>,
+    ) -> Result<Option<Page<DynamicFieldOutput>>, Error> {
```

Comment:
nit: It would be great if the `Page` type itself had a way of describing an "empty" page, so we could avoid the nesting with `Option` -- it will lead people to always have to perform two checks: Is there a page? And then are there any elements in the page? If they only care about iterating through the elements of the page, then this is a bit clunky.

Reply:
See #36